gridcomps/cap3g/Cap.F90

```diff
@@ -84,11 +94,73 @@ subroutine set_time_interval(interval, key, hconfig, rc)
       integer, optional, intent(out) :: rc
       
       integer :: status
+
+      integer :: strlen,ppos,cpos,lpos,tpos
+      integer year,month,day,hour,min,sec
+      character(len=:), allocatable :: date_string,time_string
       character(:), allocatable :: iso_duration
       
       iso_duration = ESMF_HConfigAsString(hconfig, keystring=key, _RC)
 !#      call ESMF_TimeIntervalSet(interval, timeString=iso_duration, _RC)
-      
+      year=0
+      month=0
+      day=0
+      hour=0
+      min=0
+      sec=0
+      strlen = len_trim(iso_duration)
+      tpos = index(iso_duration,'T')
+      ppos = index(iso_duration,'P')
+      _ASSERT(iso_duration(1:1) == 'P','Not valid time duration')
+
+      if (tpos /= 0) then
+         if (tpos /= ppos+1) then
+            date_string = iso_duration(ppos+1:tpos-1)
+         end if
+         time_string = iso_duration(tpos+1:strlen)
+      else
+         date_string = iso_duration(ppos+1:strlen)
+      end if
+
+      if (allocated(date_string)) then
+         strlen = len_trim(date_string)
+         lpos = 0
+         cpos = index(date_string,'Y')
+         if (cpos /= 0) then
+            read(date_string(lpos+1:cpos-1),*)year
+            lpos = cpos
+         end if
+         cpos = index(date_string,'M')
+         if (cpos /= 0) then
+            read(date_string(lpos+1:cpos-1),*)month
+            lpos = cpos
+         end if
+         cpos = index(date_string,'D')
+         if (cpos /= 0) then
+            read(date_string(lpos+1:cpos-1),*)day
+            lpos = cpos
+         end if
+      end if      
+      if (allocated(time_string)) then
+         strlen = len_trim(time_string)
+         lpos = 0
+         cpos = index(time_string,'H')
+         if (cpos /= 0) then
+            read(time_string(lpos+1:cpos-1),*)hour
+            lpos = cpos
+         end if
+         cpos = index(time_string,'M')
+         if (cpos /= 0) then
+            read(time_string(lpos+1:cpos-1),*)min
+            lpos = cpos
+         end if
+         cpos = index(time_string,'S')
+         if (cpos /= 0) then
+            read(time_string(lpos+1:cpos-1),*)sec
+            lpos = cpos
+         end if
+      end if
+      call ESMF_TimeIntervalSet(interval, yy=year, mm=month, d=day, h=hour, m=min, s=sec,_RC) 
```

Comment:
@darianboggs should have a procedure for this if ESMF does not provide it directly.  (I think we opened a ticket with them.)

Reply:
@tclune.  ESMF has not implemented this yet. If @darianboggs has a procedure then let's use that. I didn't know he did

Reply:
ESMF_TimeIntervalSet does support a ISO8601 datetime string as input. The argument is `timeIntervalString`.

Reply:
@tclune. maybe we should just keep the code I have, I know it works, the ESMF functionality is coming, just not in a release yet.
I can't seem to use anything from MAPL_BaseMod in cap3g (want to figure this too, I don't understand), but @darianboggs thought the ISO8601 that was implemented maybe not have been tested throughly. I know the code I put there works and this is just for development and can be stripped out as when ESMF has this.

Reply:
That's fine.   But open another ticket to repair this in the future.